test/Microsoft.AspNetCore.Server.Kestrel.Core.Tests/MessageBodyTests.cs

```diff
@@ -130,24 +116,79 @@ public class MessageBodyTests
 
                 count = await stream.ReadAsync(buffer, 0, buffer.Length);
                 Assert.Equal(0, count);
+            }
+        }
+
+        [Fact]
+        public async Task ReadExitsGivenIncompleteChunkedExtension()
+        {
+            using (var input = new TestInput())
+            {
+                var body = MessageBody.For(HttpVersion.Http11, new FrameRequestHeaders { HeaderTransferEncoding = "chunked" }, input.FrameContext);
+                var stream = new FrameRequestStream();
+                stream.StartAcceptingReads(body);
+
+                input.Add("5;\r\0");
 
-                await bodyTask;
+                var buffer = new byte[1024];
+                var readTask = stream.ReadAsync(buffer, 0, buffer.Length);
+
+                Assert.False(readTask.IsCompleted);
+
+                input.Add("\r\r\r\nHello\r\n0\r\n\r\n");
+
+                Assert.Equal(5, await readTask.TimeoutAfter(TimeSpan.FromSeconds(10)));
+                Assert.Equal(0, await stream.ReadAsync(buffer, 0, buffer.Length));
+            }
+        }
+
+        [Fact]
+        public async Task ReadThrowsGivenChunkPrefixGreaterThanMaxInt()
+        {
+            using (var input = new TestInput())
+            {
+                var body = MessageBody.For(HttpVersion.Http11, new FrameRequestHeaders { HeaderTransferEncoding = "chunked" }, input.FrameContext);
+                var stream = new FrameRequestStream();
+                stream.StartAcceptingReads(body);
+
+                input.Add("80000000\r\n");
+
+                var buffer = new byte[1024];
+                var ex = await Assert.ThrowsAsync<OverflowException>(async () =>
```

Comment:
This is just a test I added for existing behavior, so *technically* that's a breaking change but not really in practice.